README.md

```diff
@@ -1 +1,13 @@
-# factorial
\ No newline at end of file
+# Factorial
+Compute the factorial of a given input non-negative integer.
```

Comment:
It might be useful for students to also link to the [Wikipedia article for factorial](https://en.wikipedia.org/wiki/Factorial).

Reply:
Good idea! I added a link for further learning.